loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+* adapter — Language adapter to use for this entrypoint
+* value — Points to an implementation of the entrypoint. For the default JVM language adapter, this can be in either of the following forms:
+    * `my.package.MyClass` — A class to be instantiated and used
+    * `my.package.MyClass::thing` — A static field containing an instance of the entrypoint or a method handle for entrypoints that are functional interfaces
+
+If an entrypoint does not need to specify a language adapter other than the default language adapter, the entrypoint can be represented simply as the value string instead.
+
+
+### The `plugins` field
+| Type   | Required |
+|--------|----------|
+| Array  | False    |
+
+An array of loader plugins. A plugin is an object with the following keys:
+* adapter — Language adapter to use for this plugin
+* value — Points to an implementation of the `LoaderPlugin` interface. Can be in either of the following forms:
+    * `my.package.MyClass` — A class to be instantiated and used
+    * `my.package.MyClass::thing` — A static field containing an instance of a `LoaderPlugin`
+
+If a plugin does not need to specify a language adapter other than the default language adapter, the plugin can be represented simply as the value string instead.
+
+### The `jars` field
+| Type   | Required |
+|--------|----------|
+| Array  | False    |
+
+A list of paths to nested JAR files to load, relative to the root directory inside of the mods JAR.
+
+### The `language_adapters` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+A collection of `key: value` pairs, where each key is the namespace of a language adapter and the value is an implementation of the `LanguageAdapter` interface.
+
+### The `depends` field
+| Type   | Required |
+|--------|----------|
+| Object | False    |
+
+Defines mods that this mod will not function without.
+
+A collection of `key: value` pairs, where each key is in the form of either `mavenGroup:modId` or `modId` and each value is either a [dependency object]() or an array of dependency objects. If an array of dependency objects is provided, the dependency matches if it matches ANY of the dependency objects.
+
+#### Dependency Object
+A dependency object is made up of a version specifier and a reason for that dependency. It may be represented in any of the following ways:
+```json5
+// As a JSON object, with both available fields
+{
+    version: "*",
+    reason: "Implements the same behavior as this mod."
+}
+```
+
+```json5
+// As a JSON object, omitting the optional "reason" field
+{
+    version: "*"
+}
+```
+
+```json5
+// As a single JSON string, containing only the version specifier
+"*"
+```
+
+#### Version Specifier
+A version range specifier can make use of any of the following patterns:
+* `*` — Matches any version. Will fetch the latest version available if needed
+* `1.0.0` — Matches exactly version 1.0.0 and no other versions
```

Comment:
Rust's dependency management would interpret this as `^1.0.0` instead of `=1.00`, which I feel is a much more future-proof and robust way to handle unqualified version strings. If patch versions are used as intended, `1.0.1` should be api-compatible with anything targeting `1.0.0`. We should mirror this behavior.

Reply:
There was some earlier discussion here - not sure if that answers any of your question?

https://github.com/QuiltMC/rfcs/pull/2#discussion_r591150398

Reply:
It was here. Personally I don't think it should be interpreted as `=`, since you should have to explicitly depend on an exact version, which should be an exceedingly rare case. https://github.com/QuiltMC/rfcs/pull/2#discussion_r590915393

Reply:
I agree that it should be interpreted as `^1.0.0`, of course Minecraft (as mentioned elsewhere) could cause issues but this is up to the dev to specify properly.

Reply:
I have changed the specification to behave like `^1.0.0`.